modules/openapi-generator/src/main/java/org/openapitools/codegen/OpenAPINormalizer.java

```diff
@@ -0,0 +1,427 @@
+     *
+     * @param key   rule name
+     * @param value rule value
+     */
+    public void setRule(String key, String value) {
+        this.rules.put(key, value);
+    }
+
+    /**
+     * Normalizes the OpenAPI input, which may not perfectly conform to
+     * the specification.
+     */
+    void normalize() {
+        if (rules == null || rules.isEmpty()) {
+            return;
+        }
+
+        if (this.openAPI.getComponents() == null) {
+            this.openAPI.setComponents(new Components());
+        }
+
+        if (this.openAPI.getComponents().getSchemas() == null) {
+            this.openAPI.getComponents().setSchemas(new HashMap<String, Schema>());
+        }
+
+        normalizePaths();
+        normalizeComponents();
+    }
+
+    /**
+     * Normalizes inline models in Paths
+     */
+    private void normalizePaths() {
+        Paths paths = openAPI.getPaths();
+        if (paths == null) {
+            return;
+        }
+
+        for (Map.Entry<String, PathItem> pathsEntry : paths.entrySet()) {
+            PathItem path = pathsEntry.getValue();
+            List<Operation> operations = new ArrayList<>(path.readOperations());
+
+            // Include callback operation as well
+            for (Operation operation : path.readOperations()) {
+                Map<String, Callback> callbacks = operation.getCallbacks();
+                if (callbacks != null) {
+                    operations.addAll(callbacks.values().stream()
+                            .flatMap(callback -> callback.values().stream())
+                            .flatMap(pathItem -> pathItem.readOperations().stream())
+                            .collect(Collectors.toList()));
+                }
+            }
+
+            for (Operation operation : operations) {
+                normalizeRequestBody(operation);
+                normalizeParameters(operation);
+                normalizeResponses(operation);
+            }
+        }
+    }
+
+    /**
+     * Normalizes schemas in content
+     *
+     * @param content target content
+     */
+    private void normalizeContent(Content content) {
+        if (content == null || content.isEmpty()) {
+            return;
+        }
+
+        for (String contentType : content.keySet()) {
+            MediaType mediaType = content.get(contentType);
+            if (mediaType == null) {
+                continue;
+            } else if (mediaType.getSchema() == null) {
+                continue;
+            } else {
+                normalizeSchema(mediaType.getSchema(), new HashSet<>());
+            }
+        }
+    }
+
+    /**
+     * Normalizes schemas in RequestBody
+     *
+     * @param operation target operation
+     */
+    private void normalizeRequestBody(Operation operation) {
+        RequestBody requestBody = operation.getRequestBody();
+        if (requestBody == null) {
+            return;
+        }
+
+        // unalias $ref
+        if (requestBody.get$ref() != null) {
```

Comment:
🤔 this doesn't really fit the prevalent style of checking for `== null` first and using the `else` block for the rest.